src/components/project/overview/description.vue

```diff
@@ -23,53 +23,163 @@ except according to the terms contained in the LICENSE file.
+    lineHeight = 20;
+  }
+  return {
+    marginTop,
+    paddingTop,
+    elementHeight,
+    paddingBottom,
+    marginBottom,
+    totalHeight: marginTop + elementHeight + marginBottom,
+    lineHeight,
+    numberOfLines: Math.round(contentHeight / lineHeight)
+  };
+};
+
+const getTargetHeight = (element, allowedLines) => {
+  let linesCount = 0;
+  let runningHeight = 0;
+  let previousLineHeight = 0;
+  let previousMarginBottom = 0;
+  let previousPaddingBottom = 0;
+
+  for (const node of element.children) {
+    // Previous node reached the allowed lines and we have more node(s)
+    if (linesCount === allowedLines) {
+      return runningHeight - previousPaddingBottom - previousMarginBottom - (0.5 * previousLineHeight);
+    }
+
+    const { numberOfLines, totalHeight, marginTop, marginBottom, lineHeight, paddingTop, paddingBottom } = getExtraProperties(node);
+
+    // Only if we exceed the allowed number of lines
+    if (linesCount + numberOfLines > allowedLines) {
+      const linesToShowOfThisNode = allowedLines - linesCount - 0.5;
+      return runningHeight + marginTop + paddingTop + (linesToShowOfThisNode * lineHeight);
     }
+
+    linesCount += numberOfLines;
+    // Adjustment for margin collapse
+    runningHeight += totalHeight - Math.min(previousMarginBottom, marginTop);
+    previousLineHeight = lineHeight;
+    previousMarginBottom = marginBottom;
+    previousPaddingBottom = paddingBottom;
+  }
+  return null;
+};
+
+const setMarkdownHeight = async () => {
+  // nextTick is good enough for the calculation being done here. I don't see
+  // any screen flicker.
+  // https://github.com/vuejs/vue/issues/9200
+  await nextTick();
+  targetHeight.value = getTargetHeight(markdownRef.value.$el, 4);
+  if (targetHeight.value) {
+    markdownRef.value.$el.style.height = `${targetHeight.value}px`;
+  } else {
+    markdownRef.value.$el.style.height = 'auto';
   }
 };
+
+onMounted(async () => {
+  await setMarkdownHeight();
+});
+
+const toggle = () => {
+  if (targetHeight.value && markdownRef.value.$el.style.height === 'auto') {
+    markdownRef.value.$el.style.height = `${targetHeight.value}px`;
+  } else {
+    markdownRef.value.$el.style.height = 'auto';
+  }
+};
+
+onUpdated(async () => {
+  await setMarkdownHeight();
+});
 </script>
 
 
 <style lang="scss">
 @import '../../../assets/scss/mixins';
+@import '../../../assets/scss/variables';
 
 #project-overview-description, #project-overview-description-update {
-  border-left: 5px solid #ccc;
-  padding-left: 12px;
   margin-bottom: 20px;
 }
 
+#project-overview-description.clickable {
+  cursor: pointer;
+  border-bottom: 1px solid $color-subpanel-border;
+  overflow: hidden;
+
+  &:hover {
+    border-bottom: 1px solid $color-subpanel-border-strong;
+  }
+
+  &.expanded {
+    border-bottom: none;
+
+    &:hover {
+      border-bottom: none;
+    }
+  }
+
+}
+
 #project-overview-description {
   font-size: 16px;
   line-height: 1.2;
+  max-width: 700px;
```

Comment:
Why 700px out of curiosity?

Reply:
The max-width of a `<p>` element is usually less than this I think (at 77ch). However, @sadiqkhoja pointed out during interactive review that the project description isn't necessarily just `<p>` elements. It could have heading elements as well, which don't have a max-width.